agent/src/main/java/com/teamscale/jacoco/agent/commit_resolution/git_properties/GitPropertiesLocatorUtils.java

```diff
@@ -85,22 +87,48 @@ public static String getRevisionFromGitProperties(
 		return null;
 	}
 
+	/**
+	 * VFS (Virtual File System) protocol is used by JBoss EAP and Wildfly. Example of an URL:
+	 * vfs:/content/helloworld.war/WEB-INF/classes
+	 */
 	private static Pair<File, Boolean> getVfsContentFolder(
 			URL jarOrClassFolderUrl) throws IOException, NoSuchMethodException, IllegalAccessException, InvocationTargetException {
-		// used by JBoss EAP and Wildfly. Example: vfs:/content/helloworld.war/WEB-INF/classes
-		Object virtualFile = jarOrClassFolderUrl.openConnection().getContent();
+		// we obtain the URL of a specific class file as input, e.g.,
+		// vfs:/content/helloworld.war/WEB-INF/classes
+		// Next, we try to extract the artefact URL from it, e.g., vfs:/content/helloworld.war
+		String artefactUrl = extractArtefactUrl(jarOrClassFolderUrl);
+		if (artefactUrl == null) {
+			return null;
+		}
+
+		Object virtualFile = new URL(artefactUrl).openConnection().getContent();
 		Class<?> virtualFileClass = virtualFile.getClass();
 		// obtain the physical location of the class file. It is created on demand in <jboss-installation-dir>/standalone/tmp/vfs
 		Method getPhysicalFileMethod = virtualFileClass.getMethod("getPhysicalFile");
 		File file = (File) getPhysicalFileMethod.invoke(virtualFile);
-		if (file.getParentFile() == null || file.getParentFile().getParentFile() == null) {
+		return Pair.createPair(file, false);
+	}
+
+	private static String extractArtefactUrl(URL jarOrClassFolderUrl) {
```

Comment:
IMO it would be good to have a unit test for that

Reply:
```java.net.MalformedURLException: unknown protocol: vfs```
I cannot artificially construct a VFS URL. It's probably possible using the `org.jboss.vfs` library, but since we're doing everything with reflection now, there is no dependency anymore. I guess we don't want an extra dependency for the sake of this test case.